src/core_ocean/shared/mpas_ocn_diagnostics.F

```diff
@@ -185,6 +185,19 @@ subroutine ocn_diagnostic_solve(dt, statePool, forcingPool, meshPool, diagnostic
                 normalizedRelativeVorticityEdge, normalizedPlanetaryVorticityEdge, &
                 normalizedRelativeVorticityCell)
 
+      !
+      ! Surface pressure
+      ! This section must be placed in the code before computing the density.
+      !
+      call ocn_diagnostic_solve_surface_pressure(forcingPool, atmosphericPressure, &
+                seaIcePressure, surfacePressure)
+
+      !
+      ! Z-coordinates
+      ! This section must be placed in the code before computing the density.
+      !
+      call ocn_diagnostic_solve_z_coordinates(forcingPool, layerThickness, zMid, zTop, ssh)
```

Comment:
Does this call need to happen every time `ocn_diagnostics_solve` is called?  Or could this be wrapped in an `if ( full_compute ) then` conditional since (as far as I can tell) `zMid`, `zTop`, and `ssh` are only used in subroutines wrapped in a similar conditional.

Reply:
@mattdturner, that's a good thing to discuss.  I don't fully understand the difference between ``full_compute`` and not yet but I'll try to get caught up.

The purpose of this work is so that ``surfacePressure``, ``zMid`` and ``ssh`` can become arguments to the equation of state, and these calls appear to be outside of ``if (full_compute) then`` conditionals.  We could add another flag like
``eos_depends_on_zmid`` and then compute ``zMid`` if either ``full_compute`` or ``eos_depends_on_zmid``.  Then, as new EOSes get added, we would have to make sure ``eos_depends_on_zmid`` gets updated accurately.  Manageable but a little clumsy.

Reply:
The `full_compute` conditionals were added to try to reduce the cost of running `ocn_diagnostic_solve`, since all of the calculations in the routine are not needed each time `ocn_diagnostic_solve` is run in `mpas_ocn_time_integration_split.F`.

It looks like there is additional room for improvement, since I see before some of the calls to the routine there are notes like:
```
      ! For Richardson vertical mixing, only density,                                     
      ! layerThicknessEdge, and kineticEnergyCell need to be computed
```

For now, I would say to just leave it as is.  I will be porting the diagnostics module to GPUs soon, and will likely reassess how to effectively break up the module to allow only necessary computations to be run for each call.

Reply:
@mattdturner, looking at the one place in the code where I could find `full=.false.`:
https://github.com/MPAS-Dev/MPAS-Model/blob/ocean/develop/src/core_ocean/mode_forward/mpas_ocn_time_integration_split.F#L1708-L1711
there's a note indicating that SSH is needed even when `full=.false.`.  So that would be a pretty firm indication that we can't wrap this in `if (full_compute) then`. One could compute only ``ssh`` and not the rest but that really does sound like the point of diminishing returns to me.